sphinx-jsonschema/__init__.py

```diff
@@ -228,9 +228,9 @@ def get_json_data(self):
         elif filename and filename.startswith('http'):
             # Appears to be URL so process it as such
             schema, source = self.from_url(filename)
-        elif os.path.exists(filename):
+        elif os.path.exists(source := self._convert_filename(filename)):
```

Comment:
Keep the previous functionality of `from_file` as suggested in my other comment and pass `filename` unaltered.

Then you can replace this line with
```python
elif os.path.exists(self._convert_filename(filename)):
```
and drop the remaining two commits.

Reply:
See 21a5fd54773e1c556448a7caabbf277904d1e83a